solr/core/src/java/org/apache/solr/search/FuzzyQParserPlugin.java

```diff
@@ -0,0 +1,102 @@
+import org.apache.solr.schema.SchemaField;
+import org.apache.solr.schema.TextField;
+
+/**
+ * Plugin for parsing and creating Lucene {@link FuzzyQuery}s with all customizations available. All
+ * custom options, <code>maxEdits</code>, <code>prefixLength</code>, <code>maxExpansions</code>, and
+ * <code>transpositions</code> are optional and use the {@link FuzzyQuery} defaults if not provided.
+ *
+ * <p>Example: <code>{!fuzzy f=myfield maxEdits=1 prefixLength=3 maxExpansions=2}foobar</code>
+ *
+ * @see QueryParser
+ * @since 4.0
+ */
+public class FuzzyQParserPlugin extends QParserPlugin {
+  public static final String NAME = "fuzzy";
+
+  @Override
+  public QParser createParser(
+      String qstr, SolrParams localParams, SolrParams params, SolrQueryRequest req) {
+    return new FuzzyQParser(qstr, localParams, params, req);
+  }
+
+  static class FuzzyQParser extends QParser {
+    static final String MAX_EDITS_PARAM = "maxEdits";
+    static final String PREFIX_LENGTH_PARAM = "prefixLength";
+    static final String MAX_EXPANSIONS_PARAM = "maxExpansions";
+    static final String TRANSPOSITIONS_PARAM = "transpositions";
+
+    public FuzzyQParser(
+        String qstr, SolrParams localParams, SolrParams params, SolrQueryRequest req) {
+      super(qstr, localParams, params, req);
+    }
+
+    @Override
+    public Query parse() throws SyntaxError {
+      String termStr = getParam(QueryParsing.V);
+      String field = getParam(QueryParsing.F);
+      termStr = analyzeIfMultitermTermText(field, termStr);
+      Term t = new Term(field, termStr);
+
+      String maxEditsRaw = getParam(MAX_EDITS_PARAM);
+      int maxEdits =
+          (maxEditsRaw != null) ? Integer.parseInt(maxEditsRaw) : FuzzyQuery.defaultMaxEdits;
+      String prefixLengthRaw = getParam(PREFIX_LENGTH_PARAM);
+      int prefixLength =
+          (prefixLengthRaw != null)
+              ? Integer.parseInt(prefixLengthRaw)
+              : FuzzyQuery.defaultPrefixLength;
+      String maxExpansionsRaw = getParam(MAX_EXPANSIONS_PARAM);
+      int maxExpansions =
+          (maxExpansionsRaw != null)
+              ? Integer.parseInt(maxExpansionsRaw)
+              : FuzzyQuery.defaultMaxExpansions;
+      String transpositionsRaw = getParam(TRANSPOSITIONS_PARAM);
+      boolean transpositions =
+          (transpositionsRaw != null)
+              ? Boolean.parseBoolean(transpositionsRaw)
+              : FuzzyQuery.defaultTranspositions;
+
+      return new FuzzyQuery(t, maxEdits, prefixLength, maxExpansions, transpositions);
+    }
+
+    protected String analyzeIfMultitermTermText(String field, String part) {
+      FieldType fieldType = req.getSchema().getFieldTypeNoEx(field);
+      if (part == null
+          || !(fieldType instanceof TextField)
+          || ((TextField) fieldType).getMultiTermAnalyzer() == null) return part;
+
+      SchemaField sf = req.getSchema().getFieldOrNull((field));
+      if (sf == null || !(fieldType instanceof TextField)) return part;
```

Comment:
`!(fieldType instanceof TextField)` 
will always be false, right? The conditional check in the line above will return if this condition is true.